langchain/docstore/in_memory.py

```diff
@@ -1,17 +1,24 @@
 """Simple in memory docstore in the form of a dict."""
 from typing import Dict, Union
 
-from langchain.docstore.base import Docstore
+from langchain.docstore.base import AddableMixin, Docstore
 from langchain.docstore.document import Document
 
 
-class InMemoryDocstore(Docstore):
+class InMemoryDocstore(Docstore, AddableMixin):
     """Simple in memory docstore in the form of a dict."""
 
     def __init__(self, _dict: Dict[str, Document]):
         """Initialize with dict."""
         self._dict = _dict
 
+    def get_next_idx(self) -> int:
```

Comment:
should we just switch to using UUIDs so we dont need this?

Reply:
Hmm I ran into trouble with this approach, because at the moment the FAISS implementation matches the vector to the docstore text by integer index. So you need integer IDs.